### ISSUE TITLE
Diagnostics not shown under packages in the dependencies node

### ISSUE THREAD

**Issue by davkean** [dotnet/project-system#3780]
``` XML
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="FooBar" Version="1.0.0" />
  </ItemGroup>
</Project>

```

Expected: For the error to be listed under "FooBar" within the dependency node and for the icon to be an error.

Actual:

![image](https://user-images.githubusercontent.com/1103906/43135456-cc9aad66-8f88-11e8-93fd-6c6e8ea91681.png)

**Comment by davkean** [comment]
This regressed sometime in the last couple of releases - I'm not sure when. I think @nguerrera mentioned there was a change in the SDK - but not I'm not sure what, the errors still show up in design-time builds.

**Comment by drewnoakes** [comment]
After restoring this project, `project.assets.json` contains:

```json
  "logs": [
    {
      "code": "NU1102",
      "level": "Error",
      "message": "Unable to find package FooBar with version (>= 1.0.0)\r\n  - Found 4 version(s) in nuget.org [ Nearest version: 0.1.8.12 ]\r\n  - Found 0 version(s) in Microsoft Visual Studio Offline Packages\r\n  - Found 0 version(s) in NuGet Local\r\n  - Found 0 version(s) in dotnet.myget.org roslyn",
      "libraryId": "FooBar",
      "targetGraphs": [
        ".NETCoreApp,Version=v2.1"
      ]
    }
  ]
```

This appears in the error list:

![image](https://user-images.githubusercontent.com/350947/60576775-5dd50000-9dc1-11e9-9be0-bb59594f4916.png)

The Dependencies node is currently expecting a `ResolvedPackageReference` item (or `PackageItem`, but I don't think we'll see these during evaluation) with `ProjectItemMetadata.Type` metadata of `Diagnostic` to be provided in the `IProjectSubscriptionUpdate`. However we don't see these.

@davkean [mentions above](https://github.com/dotnet/project-system/issues/3780#issuecomment-407373916) that @nguerrera may have said (a long time ago) that there was a change to how this works. Can either of you provide some guidance so I can get this fixed?

This problem is also blocking https://github.com/dotnet/project-system/issues/2582#issuecomment-507994415.

**Comment by nguerrera** [comment]
We stopped depending on the targets that created insane numbers of ITaskItem[] for dependencies node and destroyed incremental build perf. But they still exist and project system still calls them in design time build. There’s another issue to rework the dependencies node to not depend on these targets. I’ll link it when I find it. 

I do not remember this issue. I would think that you should still get all the same items with all the same data as long as you call the same design time targets. I would not expect it to be possible to get the tree without the error items.

But it’s been a very long time since I’ve looked at this code. I’ll do some debugging.

**Comment by drewnoakes** [comment]
Thanks Nick. My research led me to `PreprocessPackageDependenciesDesignTime` in the SDK, and its corresponding invocation in `src\Tasks\Microsoft.NET.Build.Tasks\targets\Microsoft.PackageDependencyResolution.targets`.

I'm out of time for the day so didn't get much further than that. It looks like the code's still there to produce these `DependencyType.Diagnostic` task items.

**Comment by nguerrera** [comment]
So I found a couple of things near there:

1. ResolvePackageAssets hits this error and stops the DT build dead, before it gets to creating the diagnostic item. Seems this was relying on an ErrorAndContinue when the build still used the crazy DT targets. 
2. Adjusting to ErrorAndContinue on DT build produces an item,  but there doesn't seem to be a node for FooBar parenting it. 

I will dig in some more later.

**Comment by drewnoakes** [comment]
> We stopped depending on the targets that created insane numbers of ITaskItem[] for dependencies node and destroyed incremental build perf

@nguerrera I'm going to investigate whether the dependencies node can work directly with `project.assets.json` for this information. It's been on the backlog for a while and this seems like a good place to start that work.

Is there an API for parsing the file that you'd recommend?

**Comment by davkean** [comment]
@drewnoakes Let's not wait on progress for reading project.assets.json to make existing infrastructure work. We already get diagnostics from the build, we should figure out why they aren't coming through these cases.

**Comment by nguerrera** [comment]
Item 1 of 2 above is in PR to sdk (from @drewnoakes)

**Comment by nguerrera** [comment]
https://github.com/dotnet/sdk/pull/3398

**Comment by drewnoakes** [comment]
See also https://github.com/dotnet/sdk/issues/26 for history.

**Comment by nguerrera** [comment]
It's not every day that you see a two digit issue referenced

**Comment by raffaeler** [comment]
I asked from @nkolev92 to upload here my assets json file
[project.assets.json.zip](https://github.com/dotnet/project-system/files/4258076/project.assets.json.zip)
The compiler does not produce any error/warning.
No output from nuget step as well.

**Comment by nkolev92** [comment]
@raffaeler Maybe you are seeing https://github.com/dotnet/project-system/issues/2509 then, but I'm not a project-system eng, they'll know better :)

**Comment by raffaeler** [comment]
@nkolev92 It ma be the case but the main problem is over-using the yellow exclamation mark for very different errors and without any kind of hint for the developer.
Every time there is a warning, IMO it should be accompained with a detailed error and possible causes.

Who can fix this misuse, before my specfici case?

**Comment by drewnoakes** [comment]
@raffaeler thanks for the feedback. For this particular issue we have a repro, but indeed there are many reasons the yellow triangle could appear. Unfortunately we don't always know what causes a dependency to be unresolved, as a distant component could filter it out. We aim to always show something in the error list, and potentially include a tool tip (#4357).

This issue is tracking the fact that we're no longer receiving package diagnostic data, which means they're no longer appearing as child nodes in the tree.

**Comment by raffaeler** [comment]
@drewnoakes could you please use a different icon whenever the cause is different?
This is extremely confusing for everyone. 
Even if you fix this issue for the future, people will just ignore the yellow triangle 'as they did in the past'. IMO you should introduce a different marker for *known* errors and let the yellow triangle for the unfixed problem (or remote it completely as it is just useless)
